backend/stakepoold/server.go

```diff
@@ -377,18 +359,25 @@ func runMain() error {
 		}()
 	}
 
-	// Wait for CTRL+C to signal goroutines to terminate via quit channel.
+	// Wait for CTRL+C to signal goroutines to terminate
+	<-shutdownContext.Done()
```

Comment:
If my previous comment is applied, we'd only need to `ctx.Wg.Wait()` here without waiting for the `shutdownContext` to be canceled.

Actually, waiting for this `shutdownContext` to be canceled isn't relevant either ways, since we'll still have to wait for the waitgroup counter to return to 0; and each of the goroutines that decrement the waitgroup counter do so only after the `shutdownContext` is canceled.

If there's something we should do here, is make the waitgroup counter increment whenever a shutdown context is created, then ensure that each method that creates and uses a shutdown context decrements the waitgroup after performing any necessary cleanups on receiving the context cancelation signal. Then here, we  only wait for the waitgroup counter to decrement to 0.

Reply:
With the latest changes which have been pushed, I think the wait for shutdownContext is now redundant and can be removed. There is no harm in keeping it there, but it is not necessary. Do you concur @itswisdomagain @JoeGruffins ?